dev/Common/IsWindowsVersion.h

```diff
@@ -20,8 +20,17 @@ inline bool IsExportPresent(
         {
             return true;
         }
+        else
+        {
+            DWORD getProcAddressError = GetLastError();
+            if (getProcAddressError != ERROR_PROC_NOT_FOUND)
+            {
+                THROW_HR(HRESULT_FROM_WIN32(getProcAddressError));
+            }
+            return false;
+        }
     }
-    return false;
+    THROW_LAST_ERROR();
```

Comment:
Ick. Valid, but the logic's kinda spread.

If the is refactored to early-return this would be clearer.

Also, shouldn't LoadLibrary=ERROR_MOD_NOT_FOUND == false (not error)?

How about this?

```
{
    wil::unique_hmodule dll{ LoadLibraryExW(filename, nullptr, 0) };
    if (!dll)
    {
        const auto lastError{ GetLastError() };
        if (lastError == ERROR_MOD_NOT_FOUND)
        {
            return false;
        }
        THROW_HR(HRESULT_FROM_WIN32(lastError));
    }

    auto function{ GetProcAddress(dll.get(), functionName) };
    if (!function)
    {
        const auto lastError{ GetLastError() };
        if (lastError == ERROR_PROC_NOT_FOUND)
        {
            return false;
        }
        THROW_HR(HRESULT_FROM_WIN32(lastError));
    }

    return true;
}
```

Reply:
Sure, I like the refactor. But I don't agree with ERROR_MOD_NOT_FOUND. Unless there is an expected case where the module itself could not be found, it should throw an error. Is there an expected case where the kernelbase.dll wouldn't be found ?

Reply:
It futureproofs against probing for APIs in new DLLs or APIsets.